### PR TITLE
M4934 Metadata model changes

### PR DESCRIPTION
**Pull request by dennishendriksen**
See: http://www.molgenis.org/ticket/4934

Metadata entity type changes:
- Remove EntityTypeMetadata.NAME and PackageMetadata.NAME
- Update PackageMetadata.LABEL: add ROLE_LABEL and ROLE_LOOKUP, setNillable(false)
- Update EntityTypeMetadata label from "Entity" to "Entity type"
- Update EntityTypeMetadata attribute order such that it is in sync with PackageMetadata
- Removed Package.getFullyQualifiedName and EntityType.getFullyQualifiedName

Metadata entity type validation changes:
- Package, entity type and attribute name length limit was thirty characters, id length limit is 255

#### Checklist
- [ ] Functionality works & meets specifications
- [ ] Code reviewed
- [ ] Code unit/integration/system tested
- [ ] User documentation updated
- [ ] (If you have changed REST API interface) view-swagger.ftl updated
- [ ] Test plan template updated
- [ ] Clean commits

### CODE REVIEW

File: molgenis-data/src/main/java/org/molgenis/util/SecurityDecoratorUtils.java [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-	public static void validatePermission(String entityId, String entityName, Permission permission)
+	public static void validatePermission(EntityType entityType, Permission permission)
 	{
-		String role = format("ROLE_ENTITY_%s_%s", permission.toString(), entityId);
+		String role = format("ROLE_ENTITY_%s_%s", permission.toString(), entityType.getId());
 		if (!currentUserHasRole("ROLE_SU", "ROLE_SYSTEM", role))
 		{
 			throw new MolgenisDataAccessException(
-					format("No [%s] permission on entity [%s]", permission.toString(), entityName));
+					format("No [%s] permission on entity [%s]", permission.toString(), entityType.getLabel()));
```
Comment (ConnorStroomberg) [inline molgenis-data/src/main/java/org/molgenis/util/SecurityDecoratorUtils.java]: Maybe include both label and id, also update the rest-IT tests after this

**Comment by ConnorStroomberg** [comment]
When importing a emx file without a entities sheet the import fails, it should not fail.

This behaviour can be reproduced by trying to upload a emx file without entities sheet with the interface or rest-api (V1).

**Comment by ConnorStroomberg** [comment]
When i copy a entity it seems to ignore the "newEntityName" and create a uuid instead.

this is tested by: RestControllerV2APIIT.testCopyEntity

File: molgenis-data/src/main/java/org/molgenis/util/SecurityDecoratorUtils.java (hunk 1 above)
Comment (dennishendriksen) [inline molgenis-data/src/main/java/org/molgenis/util/SecurityDecoratorUtils.java]: See https://github.com/dennishendriksen/molgenis/pull/18

**Comment by Mark-de-Haan** [comment]
Closed and reopend with a merge with the master

**Comment by ConnorStroomberg** [comment]
@dennishendriksen  could you merge with Mark if needed, update with master and fix the conflicts, i will then review and merge with master if all test have passed

**Comment by ConnorStroomberg** [comment]
@dennishendriksen there is a compile error in the pull request, could you fix it and update the pull request, or let me know if you do not have time to do it.

**Comment by dennishendriksen** [comment]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] Molgenis ........................................... SUCCESS [  3.443 s]
[INFO] molgenis-test ...................................... SUCCESS [  2.961 s]
[INFO] molgenis-core ...................................... SUCCESS [  9.192 s]
[INFO] molgenis-security-core ............................. SUCCESS [  6.183 s]
[INFO] molgenis-data ...................................... SUCCESS [ 37.532 s]
[INFO] molgenis-data-cache ................................ SUCCESS [  9.763 s]
[INFO] molgenis-jobs ...................................... SUCCESS [  6.326 s]
[INFO] molgenis-data-i18n ................................. SUCCESS [  5.944 s]
[INFO] molgenis-security .................................. SUCCESS [ 12.983 s]
[INFO] molgenis-data-elasticsearch ........................ SUCCESS [ 25.697 s]
[INFO] molgenis-file ...................................... SUCCESS [  4.360 s]
[INFO] molgenis-scripts-core .............................. SUCCESS [  4.212 s]
[INFO] molgenis-js ........................................ SUCCESS [ 15.856 s]
[INFO] molgenis-data-validation ........................... SUCCESS [ 13.809 s]
[INFO] molgenis-data-postgresql ........................... SUCCESS [ 14.543 s]
[INFO] molgenis-util ...................................... SUCCESS [  6.146 s]
[INFO] molgenis-settings .................................. SUCCESS [  4.135 s]
[INFO] molgenis-data-platform ............................. SUCCESS [  0.695 s]
[INFO] molgenis-data-csv .................................. SUCCESS [  7.013 s]
[INFO] molgenis-core-ui ................................... SUCCESS [01:00 min]
[INFO] molgenis-apps ...................................... SUCCESS [  9.220 s]
[INFO] molgenis-data-excel ................................ SUCCESS [  8.768 s]
[INFO] molgenis-data-import ............................... SUCCESS [ 11.330 s]
[INFO] molgenis-data-vcf .................................. SUCCESS [  9.256 s]
[INFO] molgenis-data-annotators ........................... SUCCESS [ 22.010 s]
[INFO] molgenis-file-ingester ............................. SUCCESS [  7.061 s]
[INFO] molgenis-bootstrap ................................. SUCCESS [  0.617 s]
[INFO] molgenis-data-rest ................................. SUCCESS [ 16.734 s]
[INFO] molgenis-ontology-core ............................. SUCCESS [  7.119 s]
[INFO] molgenis-data-semanticsearch ....................... SUCCESS [  9.092 s]
[INFO] molgenis-r ......................................... SUCCESS [  0.298 s]
[INFO] molgenis-ontology .................................. SUCCESS [ 11.201 s]
[INFO] molgenis-das ....................................... SUCCESS [  6.924 s]
[INFO] molgenis-dataexplorer .............................. SUCCESS [ 12.343 s]
[INFO] molgenis-pathways .................................. SUCCESS [  8.370 s]
[INFO] molgenis-data-googlespreadsheet .................... SUCCESS [  6.828 s]
[INFO] molgenis-python .................................... SUCCESS [  0.339 s]
[INFO] molgenis-charts .................................... SUCCESS [  7.457 s]
[INFO] molgenis-data-mapper ............................... SUCCESS [ 19.899 s]
[INFO] molgenis-model-registry ............................ SUCCESS [  2.379 s]
[INFO] molgenis-data-migrate .............................. SUCCESS [  0.657 s]
[INFO] molgenis-questionnaires ............................ SUCCESS [  0.561 s]
[INFO] molgenis-catalogue ................................. SUCCESS [  0.400 s]
[INFO] molgenis-scripts ................................... SUCCESS [  0.531 s]
[INFO] molgenis-gavin ..................................... SUCCESS [ 12.898 s]
[INFO] molgenis-genetics-diag ............................. SUCCESS [  8.648 s]
[INFO] molgenis-fair ...................................... SUCCESS [  9.875 s]
[INFO] molgenis-swagger ................................... SUCCESS [  4.888 s]
[INFO] molgenis-app ....................................... SUCCESS [ 21.605 s]
[INFO] molgenis-data-rest-client .......................... SUCCESS [  4.838 s]
[INFO] molgenis-platform-integration-tests ................ SUCCESS [05:26 min]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 13:41 min
[INFO] Finished at: 2017-04-13T11:40:53+02:00
[INFO] Final Memory: 156M/856M
[INFO] ------------------------------------------------------------------------

**Comment by ConnorStroomberg** [comment]
test this please

**Pull request merged**